File: order/src/main/java/localfooddelivery/domain/Order.java
package localfooddelivery.domain;

import localfooddelivery.domain.주문됨;
import localfooddelivery.domain.Ordered;
import localfooddelivery.domain.주문취소됨;
import localfooddelivery.domain.OrderCanceled;
import localfooddelivery.OrderApplication;
import javax.persistence.*;
import java.util.List;
import lombok.Data;
import java.util.Date;

@Entity
@Table(name="Order_table")
@Data

public class Order  {

    
    @Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    
    
    
    
    
    private Long orderId;
    
    
    
    
    
    private String menu;
    
    
    
    
    
    private Integer qty;
    
    
    
    
    
    private String status;
    
    
    
    
    
    private String userId;
    
    
    
    
    
    private String address;

    @PostPersist
    public void onPostPersist(){

        //Following code causes dependency to external APIs
        // it is NOT A GOOD PRACTICE. instead, Event-Policy mapping is recommended.


        localfooddelivery.external.결제이력 결제이력 = new localfooddelivery.external.결제이력();
        // mappings goes here
        Application.applicationContext.getBean(localfooddelivery.external.결제이력Service.class)
            .결제(결제이력);


        주문됨 주문됨 = new 주문됨(this);
        주문됨.publishAfterCommit();


        //Following code causes dependency to external APIs
        // it is NOT A GOOD PRACTICE. instead, Event-Policy mapping is recommended.


        localfooddelivery.external.Payment payment = new localfooddelivery.external.Payment();
        // mappings goes here
        OrderApplication.applicationContext.getBean(localfooddelivery.external.PaymentService.class)
            .approvePayment(payment);


        Ordered ordered = new Ordered(this);
        ordered.publishAfterCommit();

    }
    @PrePersist
    public void onPrePersist(){


        주문취소됨 주문취소됨 = new 주문취소됨(this);
        주문취소됨.publishAfterCommit();



        OrderCanceled orderCanceled = new OrderCanceled(this);
        orderCanceled.publishAfterCommit();

    }
    @PreRemove
    public void onPreRemove(){
    }

    public static OrderRepository repository(){
        OrderRepository orderRepository = OrderApplication.applicationContext.getBean(OrderRepository.class);
        return orderRepository;
    }




    public static void orderStatusChange(DeliveryStarted deliveryStarted){

        /** Example 1:  new item 
        Order order = new Order();
        repository().save(order);

        */

        /** Example 2:  finding and process
        
        repository().findById(deliveryStarted.get???()).ifPresent(order->{
            
            order // do something
            repository().save(order);


         });
        */

        
    }


}

File: order/Order.py
from sqlalchemy.ext.declarative import declarative_base 
from sqlalchemy import Column, String, Integer, event, Float, Boolean
from datetime import datetime

import util
from 주문됨 import 주문됨
from external.결제이력 import 결제이력
from external.결제이력Service import 결제
from 주문취소됨 import 주문취소됨
from Ordered import Ordered
from external.Payment import Payment
from external.PaymentService import ApprovePayment
from OrderCanceled import OrderCanceled

Base = declarative_base()

class Order(Base):
    __tablename__ = 'Order_table'
    orderid = Column(Integer, primary_key=True)
    menu = Column(String(50))
    qty = Column(Integer)
    status = Column(String(50))

    def __init__(self):
        self.orderid = None
        self.menu = None
        self.qty = None
        self.status = None

@event.listens_for(Order, 'after_insert')
def PostPersist(mapper, connection, target):
    event = 주문됨()
    event = util.AutoBinding(target, event)

    event.Publish()
    
    결제이력 = 결제이력()
    response = 결제(결제이력)

    print(response)
    event = Ordered()
    event = util.AutoBinding(target, event)

    event.Publish()
    
    payment = Payment()
    response = ApprovePayment(payment)

    print(response)

    
@event.listens_for(Order, 'before_insert')
def PrePersist(mapper, connection, target):
    event = 주문취소됨()
    event = util.AutoBinding(target, event)

    event.Publish()
    
    event = OrderCanceled()
    event = util.AutoBinding(target, event)

    event.Publish()
    

    
@event.listens_for(Order, 'before_delete')
def PreRemove(mapper, connection, target):

    

